src/System.IO/src/System/IO/MemoryStream.cs

```diff
@@ -193,14 +193,16 @@ public override void Flush()
         {
         }
 
-#pragma warning disable 1998 //async method with no await operators
-        public override async Task FlushAsync(CancellationToken cancellationToken)
+        public override Task FlushAsync(CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
 
             Flush();
+            return Task.CompletedTask;
```

Comment:
This is not a correct transformation: what if Flush is overridden?  If we're going to change this, it should be made to match what's in coreclr.